src/dbnode/persist/fs/types.go

```diff
@@ -489,3 +489,22 @@ type BlockRetrieverOptions interface {
 	// IdentifierPool returns the identifierPool
 	IdentifierPool() ident.Pool
 }
+
+type ForEachRemainingFn func(seriesID ident.ID, tags ident.Tags) bool
+
+// MergeWith is an interface that the fs merger uses to merge data with.
+type MergeWith interface {
+	// Read returns the data for the given block start and series ID, whether
+	// any data was found, and the error encountered (if any).
+	Read(
+		seriesID ident.ID,
+		blockStart xtime.UnixNano,
+		nsCtx namespace.Context,
+	) ([]xio.BlockReader, bool, error)
+
+	// ForEachRemaining is the loop for the second stage of merging. The
+	// fsMerger first loops through the fileset series, merging them with data
+	// in the merge target. The second stage is the go through the merge target
+	// data and write the remaining series that were not merged.
+	ForEachRemaining(blockStart xtime.UnixNano, fn ForEachRemainingFn) error
```

Comment:
I feel like the doc string for this method should really just say that this function will loop through each combination seriesID/blockStart that was not already handled by a call to `Read()`